### ISSUE TITLE
How could I implement a loss function that was dynamic changed  in Keras

### ISSUE THREAD

**Issue by threezj** [fchollet/keras#4525]
Sorry, I am new to Keras. Now I want to implement a auto-encoder embedded k-means. But k-means's `Center point` was dynamic changed in every epoch.So how to implement it?
